core/runtime/bindings.rs

```diff
@@ -368,6 +369,45 @@ pub extern "C" fn host_initialize_import_meta_object_callback(
   let resolve_key =
     v8::String::new_external_onebyte_static(scope, b"resolve").unwrap();
   meta.set(scope, resolve_key.into(), val.into());
+
+  maybe_add_import_meta_filename_dirname(scope, meta, &name);
+}
+
+fn maybe_add_import_meta_filename_dirname(
+  scope: &mut v8::HandleScope,
+  meta: v8::Local<v8::Object>,
+  name: &str,
+) {
+  // For `file:` URL we provide additional `filename` and `dirname` values
+  let Ok(name_url) = Url::parse(name) else {
+    return;
+  };
+
+  if name_url.scheme() != "file" {
+    return;
+  }
+
+  // If something goes wrong acquiring a filepath, let skip instead of crashing
+  // (mostly concerned about file paths on Windows).
+  let Ok(mut file_path) = name_url.to_file_path() else {
+    return;
+  };
+
+  let filename_key =
+    v8::String::new_external_onebyte_static(scope, b"filename").unwrap();
+  let dirname_key =
+    v8::String::new_external_onebyte_static(scope, b"dirname").unwrap();
+  // Use display() here so that Rust takes care of proper forward/backward slash
+  // formatting depending on the OS.
+  let filename_val =
+    v8::String::new(scope, &file_path.display().to_string()).unwrap();
+
+  assert!(file_path.pop());
+  let dir_path = file_path;
+  let dirname_val =
```

Comment:
@dsherret I think this is the problematic part - for `file:///C:/some_file.js` - `pop()` here will return nothing.